src/go/rpk/pkg/cli/cmd/cloud/auth/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package auth
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Config represents the cloud configuration used to log in to the Redpanda
+// Cloud.
+type Config struct {
+	path string
+
+	ClientID     string `yaml:"client_id,omitempty"`
+	ClientSecret string `yaml:"client_secret,omitempty"`
+	AuthToken    string `yaml:"auth_token,omitempty"`
+}
+
+// defaultCfgPath returns the default path where the cloud configuration will
+// live, normally: '$HOME/.config/rpk/cloud.yaml'.
+func defaultCfgPath() (string, error) {
+	dir, err := os.UserHomeDir()
+	if err != nil {
+		return "", fmt.Errorf("unable to get home dir: %w", err)
+	}
+	return filepath.Join(dir, ".config", "rpk", "cloud.yaml"), nil
+}
+
+// LoadConfig loads the cloud configuration in the default path.
+func LoadConfig() (*Config, error) {
```

Comment:
Why not have this be a method on `Params`, which does file / env / flag processing in order?

Currently, LoadFlow first loads the file, then does env override itself. Having Params manage everything would bring this closer to the rpk Params logic, and lowers the amount that other functions need to do (LoadFlow)